volunteer/tests/test_volunteer_shift.py

```diff
@@ -0,0 +1,183 @@
+from psycopg2.errors import CheckViolation
+
+from odoo import Command
+from odoo.exceptions import ValidationError
+from odoo.tools.safe_eval import datetime
+
+from .test_volunteer_common import TestVolunteerCommon
+
+
+class TestShift(TestVolunteerCommon):
+    def setUp(self):
+        super().setUp()
+
+        # Create shifts
+        self.shift_utc_plus_1 = self.Shift.create(
+            {
+                "name": "LondonShift",
+                "state": "confirmed",
+                "start_time": datetime.datetime(2027, 6, 6, 22, 59, 59),
+                "end_time": datetime.datetime(2027, 6, 6, 23, 0, 1),
+                "tz": "Europe/London",
+                "max_volunteer_nb": 2,
+                "type_id": self.type1.id,
+            }
+        )
+        self.shift_utc_plus_2 = self.Shift.create(
+            {
+                "name": "Test",
+                "state": "confirmed",
+                "start_time": datetime.datetime(2027, 6, 6, 22, 0, 1),
+                "end_time": datetime.datetime(2027, 6, 7, 0, 0, 1),
+                "tz": "Europe/Brussels",
+                "max_volunteer_nb": 1,
+                "type_id": self.type1.id,
+            }
+        )
+
+    def test_compute_volunteer_ids(self):
+        """Test that the volunteer_ids are correctly computed
+        2 participation confirmed, 1 participation canceled"""
+
+        self.Participation.create(
+            {
+                "volunteer_id": self.volunteer_confirmed2.id,
+                "shift_id": self.shift_max_2.id,
+                "registration_state": "confirmed",
+            }
+        )
+        self.Participation.create(
+            {
+                "volunteer_id": self.volunteer_canceled.id,
+                "shift_id": self.shift_max_2.id,
+                "registration_state": "canceled",
+            }
+        )
+        self.assertEqual(
+            self.shift_max_2.volunteer_ids,
+            self.volunteer_confirmed | self.volunteer_confirmed2,
+        )
+
+    def test_is_one_day(self):
+        """Test that the shift is one day
+        start_time = 2027-06-06 23:59:58
+        end_time = 2027-06-06 23:59:59
+        shift created with tz UTC
+        expected is_one_day = True"""
+
+        shift_utc = self.Shift.create(
```

Comment:
```suggestion
        """Test if a shift is one day or not"""
        shift_utc = self.Shift.create(
```

The comment is equivalent to the code so it’s not needed.

Reply:
Done